src/components/navigation/navigation.tsx

```diff
@@ -124,12 +138,24 @@ export class CalciteNavigation {
   };
 
   private handleMenuActionSlotChange = (event: Event): void => {
-    const hasMenuAction = slotChangeHasAssignedElement(event);
-    if (hasMenuAction) {
-      this.navAction = false;
+    if (this.isPrimaryLevel(this.el)) {
+      this.navActionSlotHasElements = slotChangeHasAssignedElement(event);
+      if (this.navActionSlotHasElements) {
+        this.navAction = false;
+      }
+    }
+  };
+
+  private handleProgressSlotChange = (event: Event): void => {
+    if (this.isPrimaryLevel(this.el)) {
+      this.progressSlotHasElement = slotChangeHasAssignedElement(event);
     }
   };
 
+  private isPrimaryLevel(el: HTMLCalciteNavigationElement): boolean {
+    return el.slot !== SLOTS.navSecondary && el.slot !== SLOTS.navSecondary;
```

Comment:
Instead of passing in `this.el` into this function as an argument can you just do...

```tsx
const {el} = this;
return el.slot !== SLOTS.navSecondary && el.slot !== SLOTS.navSecondary;
```

Reply:
the problem would be scope either we have to convert this function to an arrow function of pass down the element scope from onSlotChange handler method.

Reply:
Oh ok. I was just thinking that passing the same element every time doesn't make much sense as an argument. IMO this would make more sense....

```tsx
private isPrimaryLevel = (el: HTMLCalciteNavigationElement): boolean => {
```